empiric-package/empiric/core/entry.py

```diff
@@ -1,43 +1,66 @@
-from typing import List, NamedTuple
+from __future__ import annotations
 
-from empiric.core.utils import str_to_felt
+from typing import List, Tuple, Union
 
-Entry = NamedTuple(
-    "Entry",
-    [
-        ("key", int),
-        ("value", int),
-        ("timestamp", int),
-        ("source", int),
-        ("publisher", int),
-    ],
-)
+from empiric.core.utils import str_to_felt
 
 
-def serialize_entry(entry) -> List[int]:
-    return [entry.key, entry.value, entry.timestamp, entry.source, entry.publisher]
+class Entry:
+    key: int
+    value: int
+    timestamp: int
+    source: int
+    publisher: int
```

Comment:
This is a great improvement! Turns out when you have a data structure + methods around it, a class makes things way better ;)